### PR TITLE
Added Heap Sort algorithm

### PR DESCRIPTION
**Pull request by debangi29**
## Related Issue
#805 

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

Added the Heap sort DSA algorithm
## Type of PR
- [x] Feature enhancement

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]
image

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/Sorting-Algorithms/Heap-Sort.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        }
+    }
+````
+
+  </TabItem>
+</Tabs>
+
+## 8. Complexity Analysis
+
+- Time Complexity: O(N log N)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Heap-Sort.md:278]: replace `O(N log N)` to `$O(N log N)$`

File: dsa-solutions/Sorting-Algorithms/Heap-Sort.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+- Auxiliary Space: O(log n), due to the recursive call stack. However, auxiliary space can be O(1) for iterative implementation.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Heap-Sort.md:279]: replace `O(log n)` to `$O(log n)$` and `O(1)` to `$O(1)$`

File: dsa-solutions/Sorting-Algorithms/Heap-Sort.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+## 9. Advantages and Disadvantages
+
+**Advantages:**
+- **Efficient Time Complexity**: Heap Sort has a time complexity of O(n log n) in all cases. This makes it efficient for sorting large datasets. The log n factor comes from the height of the binary heap, and it ensures that the algorithm maintains good performance even with a large number of elements.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Heap-Sort.md:284]: replace `O(n log n)` to `$O(n log n)$`

File: dsa-solutions/Sorting-Algorithms/Heap-Sort.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+- **Memory Usage** – Memory usage can be minimal (by writing an iterative heapify() instead of a recursive one). So apart from what is necessary to hold the initial list of items to be sorted, it needs no additional memory space to work
+- **Simplicity** –  It is simpler to understand than other equally efficient sorting algorithms because it does not use advanced computer science concepts such as recursion.
+
+**Disadvantages:**
+- **Costly**: Heap sort is costly as the constants are higher compared to merge sort even if the time complexity is O(n Log n) for both.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Heap-Sort.md:289]: replace `O(n Log n)` to `$O(n Log n)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by debangi29** [comment]
Please check it I have done the changes

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**